Physics/Assets/Editor/Testing Script.cs

```diff
@@ -1,116 +1,322 @@
 ﻿using ExternalUnityRendering.CameraUtilites;
-using System.Collections;
 using UnityEditor;
 using UnityEngine;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using ExternalUnityRendering.PathManagement;
+using System.Text;
+using System.Linq;
 
 namespace ExternalUnityRendering.UnityEditor
 {
-    // NOTE rework the functionality of this.
-    // also add more options
-    // TODO add sub menus for write to file, transmit etc and some way to change RUNs
-    public class TesterGUI
+    public class TesterGUI : EditorWindow
     {
-        public static int Runs = 10;
-        public static float Radius = 30;
-        public static float Power = 100;
+        private int _exportCount = 1;
+        private int _millisecondsDelay = 1000;
+        private float _explosionRadius = 1;
+        private float _explosionForce = 1;
+        private string _exportFolder = "";
+        private string _renderFolder = "";
+        private ExportScene.ExportType _exportType = ExportScene.ExportType.None;
+        private bool _exportTestRenders = false;
+        private bool _useExplosion = true;
+        private Vector3 _explosionOrigin = Vector3.zero;
+        private float _explosionUpwardsModifier = 10;
+        private ForceMode _forceType = ForceMode.Impulse;
+        private Vector2 _minMaxForce = new Vector2(1, 10);
+        private Vector2Int _renderResolution = new Vector2Int(1920, 1080);
+        private Vector2Int _rendererOutputResolution = new Vector2Int(1920, 1080);
+        private string _rendererOutputFolder = "";
```

Comment:
This default throws an error when a renderer output folder is not specified. Probably better to just ignore it, or log a warning/regular message instead.

Reply:
Added default values in init function such that no error would log. Hence there will always be an output folder.